packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4636,6 +4636,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   /// By default makes text in composing range appear as underlined.
   /// Descendants can override this method to customize appearance of text.
   TextSpan buildTextSpan() {
+    final TextStyle style = MediaQuery.boldTextOf(context)
```

Comment:
There are other codepaths in this file that talk to the platform channel about text style. I'm not clear on whether those need to be updated as well. @LongCatIsLooong or @mdebbar may know

Reply:
It looks like `SelectableText` is already doing this:

https://github.com/flutter/flutter/blob/7ce6ddabab5cee29d00fc20e2b8e4d5c99387bce/packages/flutter/lib/src/material/selectable_text.dart#L680-L682

So maybe TextField/CupertinoTextField are better places for this? Usually styles are resolved there iirc.

Reply:
Yes, `SelectableText` does this, but `EditableText` does not.

If we do it higher up in the stack, couldn't some custom text editing class end up missing out? This seems like the right peer of `SelectableText` to me.

Reply:
Ohh, I see what you're saying now.

Why not do this on `EditableText` and remove it from `SelectableText`?

Reply:
(I removed the call from SelectableText - I had seen it before but missed the part where SelectableText just returns an `EditableText` anyway)

Reply:
Ah ok. I was worried about not being able to opt-out but people can just add another MediaQuery. But I think `TextInputConnection.setTextStyle` needs to be updated as well. Also is this supposed to affect things like suffix text or placeholder text in `InputDecorator`?

Reply:
Ah ok those are `Text` widgets. But I think `TextInputConnection.setTextStyle` still needs to be updated with the right font weight if `boldText` is on.

Reply:
SG, I will do that and update the tests.

Reply:
Done